packages/core/src/hooks/useAddChain.ts

```diff
@@ -0,0 +1,98 @@
+import { AddStarknetChainParameters } from "starknet-types";
+
+import {
+  RequestArgs,
+  RequestResult,
+  UseWalletRequestProps,
+  UseWalletRequestResult,
+  useWalletRequest,
+} from "./useWalletRequest";
+
+export type AddChainVariables = Partial<AddStarknetChainParameters>;
+
+export type AddChainProps = AddChainVariables &
+  Omit<
+    UseWalletRequestProps<"wallet_addStarknetChain">,
+    keyof RequestArgs<"wallet_addStarknetChain">
+  >;
+
+export type AddChainResult = Omit<
+  UseWalletRequestResult<"wallet_addStarknetChain">,
+  "request" | "requestAsync"
+> & {
+  addChain: (args?: AddChainVariables) => void;
+  addChainAsync: (
+    args?: AddChainVariables,
+  ) => Promise<RequestResult<"wallet_addStarknetChain">>;
+};
+
+/**
+ * Hook to add a new network in the list of networks of the wallet.
+ *
+ */
+export function useAddChain(props: AddChainProps): AddChainResult {
+  const {
+    block_explorer_url,
+    chain_id,
+    chain_name,
+    icon_urls,
+    id,
+    native_currency,
+    rpc_urls,
+    ...rest
+  } = props;
+
+  let params: AddStarknetChainParameters | undefined;
+
+  if (id && chain_id && chain_name) {
+    params = {
+      block_explorer_url,
+      chain_id,
+      chain_name,
+      icon_urls,
+      id,
+      native_currency,
+      rpc_urls,
+    };
+  }
+
+  const { request, requestAsync, ...result } = useWalletRequest({
+    type: "wallet_addStarknetChain",
+    params,
+    ...rest,
+  });
+
+  const addChain = (args?: AddChainVariables) => {
+    const params_ = args ?? params;
+
+    if (!params_) throw new Error("chain arguments are required");
+    if (!params_.id) throw new Error("id is required");
+    if (!params_.chain_id) throw new Error("chain_id is required");
+    if (!params_.chain_name) throw new Error("chain_name is required");
+
+    return request({
+      params: params_ as AddStarknetChainParameters,
+      type: "wallet_addStarknetChain",
+    });
+  };
```

Comment:
In this function (like the others in this PR), if the user doesn't specify the arguments then it should just call `request()`. In that case it will call the function with the arguments provided to the hook.

In general we should not do the validation here and let the wallet return an error message if the user specify the wrong value.

```suggestion
  // equivalent to
  const addChain = (args?: AddChainVariables) => {
    return request(args ? { type: "wallet_addStarknetChain", args } : undefined);
  };
```

Reply:
Need to double check (with a demo) that it works both by calling `addChain` without any args, and by overriding the arguments.